experiment/gerrit-onboarder/onboarder.go

```diff
@@ -0,0 +1,357 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/cmd/generic-autobumper/bumper"
+)
+
+const (
+	UUID                = 0
+	GROUP_NAME          = 1
+	GROUPS_FILE         = "groups"
+	PROJECT_CONFIG_FILE = "project.config"
+
+	ACCESS_HEADER            = "[access \"refs/*\"]"
+	PROW_READ_ACCESS_FORMAT  = "read = group %s"
+	PROW_LABEL_ACCESS_FORMAT = "label-Verified = -1..+1 group %s"
+	LABEL_HEADER             = "[label \"Verified\"]"
+)
+
+type Options struct {
+	Host       string
+	Repo       string
+	UUID       string
+	GroupName  string
+	BranchName string
+}
+
+func getAllLabelLines() []string {
+	return []string{
+		"function = MaxWithBlock",
+		"value = -1 Failed",
+		"value = 0 No score",
+		"value = +1 Verified",
+		"copyAllScoresIfNoCodeChange = true",
+		"defaultValue = 0",
+	}
+}
+
+func parseAndValidateOptions() (*Options, error) {
+	var o Options
+	flag.StringVar(&o.Host, "host", "", "The gerrit host.")
+	flag.StringVar(&o.Repo, "repo", "", "The gerrit Repo.")
+	flag.StringVar(&o.UUID, "uuid", "", "The UUID to be added to the file.")
+	flag.StringVar(&o.GroupName, "group", "", "The corresponding group name for the UUID.")
+	flag.StringVar(&o.BranchName, "branch", "", "The name of the branch where refs/meta/config will live")
+	flag.Parse()
+
+	if o.Host == "" || o.Repo == "" || o.UUID == "" || o.GroupName == "" || o.BranchName == "" {
+		return &o, errors.New("all flags are required")
+	}
+
+	return &o, nil
+}
+
+func intMax(x, y int) int {
+	if x > y {
+		return x
+	}
+	return y
+}
+
+func maxIDLen(values []string) int {
+	max := 0
+	for _, item := range values {
+		max = intMax(max, len(item))
+	}
+	return intMax(max, len("# UUID"))
+}
+
+func getFormatString(maxLine int) string {
+	return "%-" + fmt.Sprintf("%d", maxLine) + "v\t%s\n"
+}
+
+func mapToGroups(groupsMap map[string]string, orderedUUIDs []string) string {
+	maxLine := maxIDLen(orderedUUIDs)
+	groups := fmt.Sprintf(getFormatString(maxLine), "# UUID", "Group Name") + "#\n"
+
+	for _, id := range orderedUUIDs {
+		groups = groups + fmt.Sprintf(getFormatString(maxLine), id, groupsMap[id])
+	}
+	return groups
+}
+
+func groupsToMap(groupsFile string) (map[string]string, []string) {
+	orderedKeys := []string{}
+	groupsMap := map[string]string{}
+	lines := strings.Split(groupsFile, "\n")
+	for _, line := range lines {
+		if !strings.HasPrefix(line, "#") && line != "" {
+			pair := strings.Split(line, "\t")
+			orderedKeys = append(orderedKeys, strings.TrimSpace(pair[UUID]))
+			groupsMap[strings.TrimSpace(pair[UUID])] = strings.TrimSpace(pair[GROUP_NAME])
+		}
+	}
+	return groupsMap, orderedKeys
+}
+
+func ensureUUID(groupsFile, id, group string) string {
+	groupsMap, orderedKeys := groupsToMap(groupsFile)
+	if value, ok := groupsMap[id]; ok && group == value {
+		return groupsFile
+	}
+	groupsMap[id] = group
+	orderedKeys = append(orderedKeys, id)
```

Comment:
Would there be duplicated keys when override?

Reply:
See above. Now erroring on dupe keys.